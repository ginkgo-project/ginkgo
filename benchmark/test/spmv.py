#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="spmv.simple.stdout",
    expected_stderr="spmv.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="spmv.simple.stdout",
    expected_stderr="spmv.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.mtx.json")],
    expected_stdout="spmv.simple.stdout",
    expected_stderr="spmv.simple.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="spmv.profile.stdout",
    expected_stderr="spmv.profile.stderr",
)
