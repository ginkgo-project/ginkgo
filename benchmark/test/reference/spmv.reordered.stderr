Running on reference(0)
Running with 2 warm iterations and 10 running iterations
The random seed for right hand sides is 42
The formats are coo
The number of right hand sides is 1
Running test case stencil(100, 7pt)
Matrix is of size (125, 125), 725
	Running spmv: coo
